sympy/solvers/solvers.py

```diff
@@ -2719,9 +2719,9 @@ def equal(expr1, expr2):
         # 2**(3*x) + x**3*log(2)**3 + 3*x**2*log(2)**2 + 3*x*log(2) + 1
         # to 2**(3*x) + (x*log(2) + 1)**3
         logs = eq.atoms(log)
-        spow = {i.exp.as_coeff_Mul()[0] for j in logs for i in j.atoms(Pow) if i.base == sym}
+        spow = {i.exp for j in logs for i in j.atoms(Pow) if i.base == sym}
         spow = min(spow) if spow else 1
```

Comment:
this will fail with symbolic exponents...what was the failing expression that got here from dsolve?

Reply:
This is the failing expression - `(x + sqrt(-x*f(x) + f(x)**2))*diff(f(x), x) - f(x)`

Reply:
I think this diff should work for symbolic exponents as well, correct me if I'm wrong.
```diff
diff --git a/sympy/solvers/solvers.py b/sympy/solvers/solvers.py
index 07ac28db2d..d5c677ad6f 100644
--- a/sympy/solvers/solvers.py
+++ b/sympy/solvers/solvers.py
@@ -2719,7 +2719,7 @@ def equal(expr1, expr2):
         # 2**(3*x) + x**3*log(2)**3 + 3*x**2*log(2)**2 + 3*x*log(2) + 1
         # to 2**(3*x) + (x*log(2) + 1)**3
         logs = eq.atoms(log)
-        spow = {i.exp for j in logs for i in j.atoms(Pow) if i.base == sym}
+        spow = {i.exp.as_coeff_Mul()[1] for j in logs for i in j.atoms(Pow) if i.base == sym}
         spow = min(spow) if spow else 1
         if eq.subs(sym**spow, Dummy()).has_free(sym):
             spow = 1  # there was a free-symbol or non-pow generator with sym
```

Reply:
```python
>>> sym=x
>>> logs = log(x**2), log(x**y)
>>> spow = {i.exp.as_coeff_Mul()[1] for j in logs for i in j.atoms(Pow) if i.base == sym}
>>> spow
{1,y}
>>> min(spow)
Traceback (most recent call last):
...
TypeError: cannot determine truth value of Relational
```

Reply:
> This is the failing expression

Since there are no log terms, I don't understand why this is failing....